### ISSUE TITLE
QHTTP modifies POST boundary

### ISSUE THREAD

**Issue by HappySeaFox** [azadkuh/qhttp#32]
Hi. I use a great QHTTP library as a HTTP server implementation. When I receive a POST multipart/form-data request I get its boundary from the "content-type" header. However I noticed, that the HTTP header value is altered and is modified to be in a lower case (HTTP standard violation?). Because of that I cannot find the received boundary in a POSTed data.

**Comment by rv-17** [comment]
Re: header fields: ”Each header field consists of a name followed by a colon (":") and the field value. Field names are case-insensitive.”

**Comment by HappySeaFox** [comment]
Correct, field names are case-insensitive. But field values are not case-insensitive

**Comment by mrdeveloperdude** [comment]
I was just going to file a report about this issue!

Long story short, to get it to work simply remove .toLower() for all header values in the qhttp code and it will work as expected (.toLower() on header key is fine as is).

And for you guys who don't understand the problem that is being addressed, here is an attempt to explain it:

When doing HTTP POST with "multipart/form-data" to upload a file with some parameters the client will prepare a header in the request like this:

`Content-Type: multipart/form-data; boundary=SOME_RANDOM_STRING\r\n`

The POST body will then contain the `SOME_RANDOM_STRING` between each data item as a means for the server to separate them into parts.

Any file uploads with all modern browsers will use this method.

As an example, the boundary string that Google Chrome generates when posting a file looks like this: "----WebKitFormBoundarySbQGNFBMdCG2vf0J". Notice that this boundary string **contains mixed case data**! If the header value is made toLower(), this boundary string will be useless, because the boundary string used to separate parts of the POST body will still be mixed case. I don't know if it is a standard violation or not, but it sure as hell breaks a pretty basic feature of HTTP POST- namely the use of boundaries to separate the fields of the uploaded data:-)

**Comment by kaedwen** [comment]
stumbled over the same issue right now why is this not gonna be fixed? It's as simple as removing the toLower() and it does not break any convention.

Matching the boundary case insensitive might be a solution too.
